ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -165,6 +165,10 @@ impl<'a> Segment<'a> {
         }
     }
 
+    /// This function returns the length of the first entry, i.e. `x_left`
+    ///
+    /// The lengths of the entries might be inconsistent, for asserting that the length is identical
+    /// across entries, call function `assert_len`
```

Comment:
I think the comments on lines 170-171, 369, and 404, are no longer applicable in the latest version.